### DISCUSSION TITLE
Simulation (.csc) files

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#1704] [Q&A]
When I want to try something quick without potentially messing my work in a certain directory, I will copy the directory and try it out there. I found out however, that the Cooja simulation definition files (.csc) have absolute paths so in the new directory it seems that even though the code have changed, Cooja will compile and run the old code, from the original directory.

Is there a way to prevent that (other than manually editing the .csc files)?

**Comment by alexrayne** [comment]
as i understand, cooja saves path that provided for source as is. open file dialog provides absolute path for source, but you can type-in any path you want. Try reconfigure project and change path to relative one.

**Comment by atiselsts** [comment]
You can use `[CONFIG_DIR]` inside paths in a .csc file, this allows to set paths relative to the directory of the .csc file.

**Reply by rahav** [reply]
I poked around the documentation and couldn't find a reference to `[CONFIG_DIR]`. Where do I define it?

**Reply by alexrayne** [reply]
there is no any doc about it. You can find it only examining sources: cooja.java:createConfigRelativePath(...)

**Reply by pjonsson** [reply]
> I poked around the documentation and couldn't find a reference to `[CONFIG_DIR]`. Where do I define it?

The Contiki-NG repository contains a lot of csc files that use the variable, `examples/benchmarks/result-visualization/cooja.csc` is one example.
